pkg/kubelet/server/remotecommand/exec.go

```diff
@@ -51,8 +85,19 @@ func ServeExec(w http.ResponseWriter, req *http.Request, executor Executor, podN
 
 	err := executor.ExecInContainer(podName, uid, container, cmd, ctx.stdinStream, ctx.stdoutStream, ctx.stderrStream, ctx.tty, ctx.resizeChan)
 	if err != nil {
-		msg := fmt.Sprintf("error executing command in container: %v", err)
-		runtime.HandleError(errors.New(msg))
-		fmt.Fprint(ctx.errorStream, msg)
+		m := ExitError{}
```

Comment:
K, new protocol!

On Wed, Aug 3, 2016 at 3:48 PM, Andy Goldstein <notifications@github.com>
wrote:

> In pkg/kubelet/server/remotecommand/exec.go
> <https://github.com/kubernetes/kubernetes/pull/26541#discussion_r73407352>
> :
>
> > @@ -51,8 +85,19 @@ func ServeExec(w http.ResponseWriter, req *http.Request, executor Executor, podN
> >
> >  	err := executor.ExecInContainer(podName, uid, container, cmd, ctx.stdinStream, ctx.stdoutStream, ctx.stderrStream, ctx.tty, ctx.resizeChan)
> >  	if err != nil {
> > -		msg := fmt.Sprintf("error executing command in container: %v", err)
> > -		runtime.HandleError(errors.New(msg))
> > -		fmt.Fprint(ctx.errorStream, msg)
> > +		m := ExitError{}
>
> Yeah, I think an old client/new server combo would be busted.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26541/files/0a10e2dcdaf86a90b5a718e96d44cc9e5286ca06#r73407352>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p-pPdIwS6AgLL7f0jCQxkMcTXP2oks5qcPCZgaJpZM4Ip8Gm>
> .
>